dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/AbstractCommandExecutor.java

```diff
@@ -207,17 +203,10 @@ public CommandExecuteResult run(String execCommand) throws Exception {
 
         // if SHELL task exit
         if (status) {
-            // set appIds
-            List<String> appIds = getAppIds(taskExecutionContext.getLogPath());
-            result.setAppIds(String.join(Constants.COMMA, appIds));
 
             // SHELL task state
             result.setExitStatusCode(process.exitValue());
 
-            // if yarn task , yarn state is final state
-            if (process.exitValue() == 0) {
-                result.setExitStatusCode(isSuccessOfYarnState(appIds) ? EXIT_CODE_SUCCESS : EXIT_CODE_FAILURE);
-            }
         } else {
             logger.error("process has failure , exitStatusCode:{}, processExitValue:{}, ready to kill ...",
                  result.getExitStatusCode(), process.exitValue());
```

Comment:
We may also need to modify the kill method on line 214. In the current implementation, it will find `appliacationId` from log and kill the application on yarn, for normal tasks, we don't need to kill on yarn.

Reply:
just remove the  killYarnJob call from kill method maybe  enough .
```java
  public static void kill(TaskExecutionContext taskExecutionContext) {
      try {
          int processId = taskExecutionContext.getProcessId();
          if (processId == 0) {
              logger.error("process kill failed, process id :{}, task id:{}",
                      processId, taskExecutionContext.getTaskInstanceId());
              return;
          }

          String pidsStr = getPidsStr(processId);
          if (StringUtils.isNotEmpty(pidsStr)) {
              String cmd = String.format("kill -9 %s", pidsStr);
              cmd = OSUtils.getSudoCmd(taskExecutionContext.getTenantCode(), cmd);
              logger.info("process id:{}, cmd:{}", processId, cmd);
              OSUtils.exeCmd(cmd);
          }

      } catch (Exception e) {
          logger.error("kill task failed", e);
      }
      // find log and kill yarn job
      killYarnJob(taskExecutionContext);
 }
```
I found  the TaskExecuteThread would call the cancelApplication to kill the yarn job  when AbstractYarnTask handle  method throws Exception.

Reply:
the cancelApplication method  of  AbstractYarnTask  would call ProcessUtils.killYarnJob(taskExecutionContext)  for itself
```java
    public void cancelApplication(boolean status) throws Exception {
        cancel = true;
        // cancel process
        shellCommandExecutor.cancelApplication();
        TaskInstance taskInstance = processService.findTaskInstanceById(taskExecutionContext.getTaskInstanceId());
        if (status && taskInstance != null) {
            ProcessUtils.killYarnJob(taskExecutionContext);
        }
    }
```

Reply:
@skymsg In some case, the task may just exit with false status and won't throw an exception.